test/test_EDM4hepFile.py

```diff
@@ -63,9 +93,20 @@ def reader(inputfile_name):
 
 
 @pytest.fixture(scope="module")
-def events(reader):
+def edm4hep_version(reader):
+    return reader.current_file_version("edm4hep")
+
+
+@pytest.fixture(scope="module")
+def events(reader, edm4hep_version):
     """Get the events from the reader"""
-    return reader.get("events")
+    collections = COLLECTION_NAMES
+    if edm4hep_version < podio.version.parse("0.99.3"):
+        collections.remove("GeneratorEventParametersCollection")
+        collections.remove("UserDataCollectionFloat")
+        collections.remove("UserDataCollectionInt")
```

Comment:
Can we not read all the collections? Maintaining a list of collections is not great